barretenberg/ts/src/barretenberg_wasm/fetch_code/node/index.ts

```diff
@@ -16,7 +16,18 @@ function getCurrentDir() {
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 export async function fetchCode(multithreaded: boolean, wasmPath?: string) {
   const path = wasmPath ?? getCurrentDir() + '/../../barretenberg-threads.wasm.gz';
-  const compressedData = await readFile(path);
-  const decompressedData = pako.ungzip(new Uint8Array(compressedData));
-  return decompressedData.buffer;
+  const maybeCompressedData = await readFile(path);
+  const buffer = new Uint8Array(maybeCompressedData);
+  const isGzip =
+    // Check magic number
+    buffer[0] === 0x1f &&
+    buffer[1] === 0x8b &&
+    // Check compression method:
+    buffer[2] === 0x08;
+  if (isGzip) {
+    const decompressedData = pako.ungzip(buffer);
+    return decompressedData.buffer;
+  } else {
+    return buffer;
+  }
```

Comment:
Why would the gz file not be gzipped? If there's a reasonable explanation it should likely be in a comment here

Reply:
If you use a different WASM_PATH, you could point it to a plain .wasm file. I'll add a comment